src/cli_driver.rs

```diff
@@ -449,14 +448,28 @@ impl ProcessingSession {
             };
 
             if let Some(par) = tex_path.parent() {
-                fs_root = par;
+                output_path = par;
                 io_builder.filesystem_root(par);
             } else {
                 return Err(ErrorKind::Msg(format!("can't figure out a parent directory for input path \"{}\"",
                                                   tex_path.to_string_lossy())).into());
             }
         }
 
+        if let Some(dir) = args.value_of("outdir") {
+
+            output_path = {
+                let tmp = Path::new(dir);
+                match tmp.extension() {
+                    Some(_) => {
+                        tt_warning!(status, "file extension in `-o/--outdir`; only the directory will be used");
+                        tmp.parent().unwrap()
+                        }
+                    None => &tmp,
+                }};
+            if !output_path.exists() { return Err(ErrorKind::Msg(format!("output directory \"{}\" does not exist", output_path.display())).into());}
```

Comment:
Please don't condense the bodies of `if` statements to single lines. That is, please put the `return` statement and closing bracket on their own lines.